libs/remix-ui/static-analyser/src/lib/remix-ui-static-analyser.tsx

```diff
@@ -0,0 +1,383 @@
+    msg: '',
+    options: {},
+    hasWarning: false,
+    warningErrors: []
+  })
+  const [warningState, setWarningState] = useState([])
+
+  useEffect(() => {
+    
+    if (autoRun) {
+      const setCompilationResult = async (data, source, file) => {
+        await setResult({ lastCompilationResult: data, lastCompilationSource: source, currentFile: file })
+      }
+      
+      if (props.analysisModule) {
+       
+        props.analysisModule.on(
+          'solidity',
+          'compilationFinished',
+          (file, source, languageVersion, data) => {
+            if (languageVersion.indexOf('soljson') !== 0) return
+            setCompilationResult(data, source, file)
+            if(categoryIndex.length > 0){
+              run(data, source, file)
+            }
+
+          }
+        )
+      }
+    } else {
+      setAutoRun(true)
+    }
+
+    return () => { }
+  }, [autoRun, categoryIndex])
+
+  const run = (lastCompilationResult, lastCompilationSource, currentFile) => {
+    // const highlightLocation = async (location, fileName) => {
+    //   await props.analysisModule.call('editor', 'discardHighlight')
+    //   await props.analysisModule.call('editor', 'highlight', location, fileName)
+    // }
+    setResult({ lastCompilationResult, lastCompilationSource, currentFile })
+    if (lastCompilationResult && categoryIndex.length) {
+      setRunButtonState(false)
+      let warningCount = 0
+      const warningMessage = []
+
+      runner.run(lastCompilationResult, categoryIndex, results => {
+        results.map((result) => {
+          let moduleName
+          Object.keys(groupedModules).map(key => {
+            groupedModules[key].forEach(el => {
+              if (el.name === result.name) {
+                moduleName = groupedModules[key][0].categoryDisplayName
+              }
+            })
+          })
+          setModuleNameResult(moduleName)
+          const warningErrors = []
+          result.report.map((item) => {
+            let location: any = {}
+            let locationString = 'not available'
+            let column = 0
+            let row = 0
+            let fileName = currentFile
+            if (item.location) {
+              var split = item.location.split(':')
+              var file = split[2]
+              location = {
+                start: parseInt(split[0]),
+                length: parseInt(split[1])
+              }
+              location = props.analysisModule._deps.offsetToLineColumnConverter.offsetToLineColumn(
+                location,
+                parseInt(file),
+                lastCompilationSource.sources,
+                lastCompilationResult.sources
+              )
+              row = location.start.line
+              column = location.start.column
+              locationString = row + 1 + ':' + column + ':'
+              fileName = Object.keys(lastCompilationResult.contracts)[file]
+            }
+            warningCount++
+            const msg = `
```

Comment:
Can we use a different implementation for this and not use string to represent JSX?

Reply:
Extracted string into a function